Mathlib/Probability/ProductMeasure.lean

```diff
@@ -403,6 +406,17 @@ lemma infinitePi_pi {s : Finset ι} {t : (i : ι) → Set (X i)}
   · exact measurable_restrict _
   · exact .univ_pi fun i ↦ mt i.1 i.2
 
+-- TODO: add a version for infinite `ι`. See TODO on `infinitePi_pi`.
+@[simp]
+lemma infinitePi_singleton [Fintype ι] [∀ i, MeasurableSingletonClass (X i)] (f : ∀ i, X i) :
+    infinitePi μ {f} = ∏ i, μ i {f i} := by
+  simpa [Set.univ_pi_singleton] using
+    infinitePi_pi μ (s := .univ) (t := fun i ↦ {f i}) fun _ _ ↦ .singleton _
+
+@[simp] lemma infinitePi_dirac [Countable ι] [∀ i, MeasurableSingletonClass (X i)] (f : ∀ i, X i) :
+    infinitePi (fun i ↦ dirac (f i)) = dirac f :=
+  .symm <| eq_infinitePi _ fun s t ht ↦ by simp [dirac_apply]
```

Comment:
```suggestion
@[simp] lemma infinitePi_dirac (f : ∀ i, X i) :
    infinitePi (fun i ↦ dirac (f i)) = dirac f :=
  .symm <| eq_infinitePi _ fun s t ht ↦ by
    rw [dirac_apply']
    swap
    · apply MeasurableSet.pi
      · exact countable_toSet s
      simp [ht]
    simp only [Set.indicator_pi_one_apply]
    congr! with i hi
    rw [dirac_apply']
    exact ht i
```
I expect you can write this proof in a much shorter way, but you can do it without both of these typeclass assumptions, which means that two of your lemmas in the binomial random PR don't need countability